src/components/DataTypes/array.tsx

```diff
@@ -49,166 +50,132 @@ interface StringFilter extends JSONSchema {
-			title || field,
-			operators[operator].getLabel(schema),
-			value,
-		),
-		type: 'object',
-	};
-
-	const filter = getFilter(schema, value);
+	const base = getBase(
+		recursive,
+		operator,
+		title,
+		field,
+		value,
+		refScheme,
+	) as ArrayFilter;
+	const filter = getFilter(field, schema, value, operator, isFullTextSearch);
 
-	if (operator === 'contains') {
-		return Object.assign(base, {
-			properties: {
-				[field]: {
-					type: 'array',
-					...filter,
-				},
+	return Object.assign(base, {
+		properties: {
+			[field]: {
+				type: 'array',
+				...filter,
 			},
-			required: [field],
-		});
-	}
-
-	if (operator === 'not_contains') {
-		return Object.assign(base, {
-			anyOf: [
-				{
-					properties: {
-						[field]: {
-							type: 'array',
-							not: {
-								...filter,
-							},
-						},
-					},
-				},
-				{
-					properties: {
-						[field]: {
-							type: 'null',
-						},
-					},
-				},
-			],
-		});
-	}
-
-	return base;
-};
-
-interface OneOf {
-	const: string | boolean | number;
-	title: string;
-}
-
-export const Edit = ({
-	onUpdate,
-	slim,
-	...props
-}: DataTypeEditProps &
-	TextareaProps &
-	InputProps &
-	Omit<SelectProps<OneOf>, 'onChange'> & { slim?: boolean }) => {
-	const schemaItems = props.schema.items as JSONSchema | undefined;
-	if (schemaItems?.oneOf) {
-		return (
-			<Select<OneOf>
-				{...props}
-				options={schemaItems.oneOf || []}
-				valueKey="const"
-				labelKey="title"
-				value={
-					(schemaItems.oneOf || []).find(
-						(x: OneOf) => x.const === props.value,
-					) as OneOf
-				}
-				onChange={({ option }) => onUpdate(option.const.toString())}
-			/>
-		);
-	}
-	if (slim) {
-		return (
-			<Input
-				onChange={(e: React.FormEvent<HTMLInputElement>) =>
-					onUpdate(e.currentTarget.value)
-				}
-				{...props}
-			/>
-		);
-	}
-
-	return (
-		<Textarea
-			onChange={(e: React.FormEvent<HTMLTextAreaElement>) =>
-				onUpdate(e.currentTarget.value)
-			}
-			{...props}
-		/>
-	);
+		},
+		required: [field],
+	});
 };
 
-export const getFilter = (schema: JSONSchema, value: string) => {
+export const getFilter = (
+	field: string,
+	schema: JSONSchema,
+	value: string,
+	operator: Operator<OperatorSlug>,
+	isFullTextSearch?: boolean,
+) => {
 	if (
 		!!schema?.items &&
 		typeof schema.items !== 'boolean' &&
 		'properties' in schema.items &&
 		!!schema.items.properties
 	) {
+		const propertyFilters = Object.entries(schema.items.properties).map(
+			([propKey, propValue]: [string, JSONSchema]) => {
+				const subSchema = isFullTextSearch
+					? { ...propValue, oneOf: undefined, enum: undefined }
+					: propValue;
+				const model = getDataModel(subSchema);
+				const operators = model?.operators(schema);
+				const newOperator = convertOperator(operator, operators);
+				const filter = (model?.createFilter as CreateFilter)(
+					propKey,
+					newOperator,
+					value,
+					subSchema,
+					true,
+				);
+				return filter;
```

Comment:
At this point, isn't this just v ?
```suggestion
				return getModelFilter(
					propValue,
					propKey,
					operator,
					value,
					isFullTextSearch,
					convertOperator,
				);
```